test/join.jl

```diff
@@ -371,4 +371,21 @@ module TestJoin
         @test levels(join(A, B, on=:b, kind = :semi)[:b]) == ["d", "c", "b", "a"]
         @test levels(join(B, A, on=:b, kind = :semi)[:b]) == ["a", "b", "c"]
     end
+
+    @testset "join on columns with different left/right names" begin
```

Comment:
Can you also tests different kinds of joins?

Reply:
Great idea. Your suspicion
> Looks mostly good, but I wonder how it can work given that left_on isn't used anywhere...

was very astute : )